core-web/libs/sdk/uve/src/lib/style-editor/types.ts

```diff
@@ -0,0 +1,616 @@
+ * All specific field types must include these base properties. The `type` property
+ * serves as a discriminator that allows TypeScript to narrow union types based on
+ * the field type.
+ *
+ * @property type - The type of field (discriminator for union types, enables type narrowing)
+ * @property label - The human-readable label displayed for this field in the UI
+ */
+export interface StyleEditorBaseField {
+    /** The type of field, used to discriminate between different field types in union types */
+    type: StyleEditorFieldType;
+    /** The label text displayed to users for this field */
+    label: string;
+}
+
+/**
+ * Input field definition with type-safe defaultValue based on inputType.
+ *
+ * This is a discriminated union type that ensures type safety between
+ * the `inputType` and `defaultValue` properties. TypeScript will enforce
+ * that the `defaultValue` type matches the `inputType`:
+ * - When `inputType` is `'number'`, `defaultValue` must be a `number`
+ * - When `inputType` is `'text'`, `defaultValue` must be a `string`
+ *
+ * This type safety prevents runtime errors and ensures that numeric inputs
+ * receive numeric values and text inputs receive string values.
+ *
+ * @example
+ * ```typescript
+ * // Number input with number defaultValue
+ * const numberField: StyleEditorInputField = {
+ *   type: 'input',
+ *   label: 'Font Size',
+ *   inputType: 'number',
+ *   defaultValue: 16, // TypeScript ensures this is a number
+ *   placeholder: 'Enter font size'
+ * };
+ *
+ * // Text input with string defaultValue
+ * const textField: StyleEditorInputField = {
+ *   type: 'input',
+ *   label: 'Font Name',
+ *   inputType: 'text',
+ *   defaultValue: 'Arial', // TypeScript ensures this is a string
+ *   placeholder: 'Enter font name'
+ * };
+ *
+ * // TypeScript error: type mismatch
+ * // const invalid: StyleEditorInputField = {
+ * //   type: 'input',
+ * //   label: 'Size',
+ * //   inputType: 'number',
+ * //   defaultValue: '16' // Error: string not assignable to number
+ * // };
+ * ```
+ */
+export type StyleEditorInputField =
+    | (StyleEditorBaseField & {
+          /** Discriminator: must be 'input' */
+          type: 'input';
+          /** Input type for number values */
+          inputType: 'number';
+          /** Optional placeholder text shown when the input is empty */
+          placeholder?: string;
+          /** Optional default value (must be a number when inputType is 'number') */
+          defaultValue?: number;
+      })
+    | (StyleEditorBaseField & {
+          /** Discriminator: must be 'input' */
+          type: 'input';
+          /** Input type for text/string values */
+          inputType: 'text';
+          /** Optional placeholder text shown when the input is empty */
+          placeholder?: string;
+          /** Optional default value (must be a string when inputType is 'text') */
+          defaultValue?: string;
+      });
```

Comment:
I think the first `|` is not needed and also we should make use of the types we have above instead of pure strings